### PR TITLE
Update datasets to clean up legacy code

### PR DESCRIPTION
**Pull request by vijayphoenix**
* Issue Reference: https://github.com/tensorflow/datasets/issues/1519

### CODE REVIEW

**Comment by vijayphoenix** [comment]
@Conchylicultor please review these changes.

I will address other parts of the issue soon.
Thank you

File: tensorflow_datasets/image/oxford_iiit_pet.py [hunk 1]
```diff
@@ -102,7 +102,6 @@ def _split_generators(self, dl_manager):
     # Setup train and test splits
     train_split = tfds.core.SplitGenerator(
         name="train",
-        num_shards=_NUM_SHARDS,
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/oxford_iiit_pet.py]: Note that you could also remove `_NUM_SHARDS = 123` at the top of the file, which isn't used anymore.

Comment (vijayphoenix) [reply tensorflow_datasets/image/oxford_iiit_pet.py]: Yeah, I noticed that.

**Comment by vijayphoenix** [comment]
Yeah, will do them in some time.

**Comment by vijayphoenix** [comment]
@Conchylicultor I think now this PR completely addresses the issue.
Thank you

**Comment by Conchylicultor** [comment]
Awesome work! Many thanks. I've sent this for review and i'll submit this first as this is the most complete.
Let's commit this first

**Comment by vijayphoenix** [comment]
FYI, only the groove dataset was not cleaned up as it was cleaned up by your Previous PR

**Pull request merged**